sphinx_gallery/gen_gallery.py

```diff
@@ -477,21 +477,29 @@ def generate_gallery_rst(app):
 
         # Create section rst files and fetch content which will
         # be added to current index file
-        this_fhindex, this_costs, this_toctree, _ = generate_dir_rst(
+        (
+            _,
+            this_content,
+            this_costs,
+            this_toctree_items,
+        ) = generate_dir_rst(
             examples_dir_abs_path,
             gallery_dir_abs_path,
             gallery_conf,
-            seen_backrefs
+            seen_backrefs,
+            include_toctree=False,
```

Comment:
And then here you do `return_toctree_filenames=True` and change the items that get returned above